File: trunk/esper/src/test/java/net/esper/view/TestViewFactory.java
package net.esper.view;

import java.util.LinkedList;
import java.util.List;

import junit.framework.TestCase;
import net.esper.support.bean.SupportMarketDataBean;
import net.esper.support.view.SupportBeanClassView;
import net.esper.support.view.SupportViewContextFactory;
import net.esper.support.event.SupportEventAdapterService;
import net.esper.type.StringValue;

import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;

public class TestViewFactory extends TestCase
{
    private SupportBeanClassView parentViewable = new SupportBeanClassView(SupportMarketDataBean.class);

    public void testCreate() throws Exception
    {
        List<Object> parameters = new LinkedList<Object>();
        parameters.add("price");
        ViewSpec spec = new ViewSpec(ViewEnum.UNIVARIATE_STATISTICS.getNamespace(), ViewEnum.UNIVARIATE_STATISTICS.getName(), parameters);

        Viewable view = ViewFactory.create(parentViewable, spec);
        if (view instanceof ContextAwareView)
        {
            ContextAwareView contextAwareView = (ContextAwareView) view;
            contextAwareView.setViewServiceContext(SupportViewContextFactory.makeContext());            
        }

        assertTrue(view != null);
        assertTrue(view.getViews().size() == 0);
        assertTrue(view.getEventType().getPropertyType(ViewFieldEnum.UNIVARIATE_STATISTICS__SUM.getName()) != null);
    }

    public void testInvalidViewName()
    {
        ViewSpec spec = new ViewSpec("dummy", "bumblebee", null);

        try
        {
            ViewFactory.create(parentViewable, spec);
            assertFalse(true);
        }
        catch (ViewProcessingException ex)
        {
            log.debug(".testInvalidViewName Expected exception caught, msg=" + ex.getMessage());
        }
    }

    public void testInvalidViewParameters()
    {
        // Forget to populate view parameters, should error
        List<Object> parameters = new LinkedList<Object>();
        ViewSpec spec = new ViewSpec("stat", "uni", parameters);

        try
        {
            ViewFactory.create(parentViewable, spec);
            assertFalse(true);
        }
        catch (ViewProcessingException ex)
        {
            log.debug(".testInvalidViewParameters Expected exception caught, msg=" + ex.getMessage());
        }
    }

    public void testInvalidAttach()
    {
        // View doesn't allow non-numeric fields, the factory should check this
        List<Object> parameters = new LinkedList<Object>();
        parameters.add(new StringValue("symbol"));
        ViewSpec spec = new ViewSpec("stat", "uni", parameters);

        try
        {
            ViewFactory.create(parentViewable, spec);
            assertFalse(true);
        }
        catch (ViewProcessingException ex)
        {
            log.debug(".testInvalidViewName Expected exception caught, msg=" + ex.getMessage());
        }
    }

    private static final Log log = LogFactory.getLog(TestViewFactory.class);
}

File: NEsper.Tests/view/TestViewFactory.cs
using System;
using System.Collections.Generic;

using net.esper.support.bean;
using net.esper.support.events;
using net.esper.support.view;
using net.esper.type;

using NUnit.Core;
using NUnit.Framework;

using org.apache.commons.logging;

namespace net.esper.view
{

	[TestFixture]
    public class TestViewFactory
    {
        private SupportBeanClassView parentViewable = new SupportBeanClassView(typeof(SupportMarketDataBean));

        [Test]
        public virtual void testCreate()
        {
            IList<Object> parameters = new List<Object>();
            parameters.Add("price");
            ViewSpec spec = new ViewSpec(ViewEnum.UNIVARIATE_STATISTICS.Namespace, ViewEnum.UNIVARIATE_STATISTICS.Name, parameters);

            Viewable view = ViewFactory.Create(parentViewable, spec);
            if (view is ContextAwareView)
            {
                ContextAwareView contextAwareView = (ContextAwareView)view;
                contextAwareView.ViewServiceContext = SupportViewContextFactory.makeContext();
            }

            Assert.IsTrue(view != null);
            Assert.IsTrue(view.Views.Count == 0);
            Assert.IsTrue(view.EventType.GetPropertyType(ViewFieldEnum.UNIVARIATE_STATISTICS__SUM.Name) != null);
        }

        [Test]
        public virtual void testInvalidViewName()
        {
            ViewSpec spec = new ViewSpec("dummy", "bumblebee", null);

            try
            {
                ViewFactory.Create(parentViewable, spec);
                Assert.IsFalse(true);
            }
            catch (ViewProcessingException ex)
            {
                log.Debug(".testInvalidViewName Expected exception caught, msg=" + ex.Message);
            }
        }

        [Test]
        public virtual void testInvalidViewParameters()
        {
            // Forget to populate view parameters, should error
            IList<Object> parameters = new List<Object>();
            ViewSpec spec = new ViewSpec("stat", "uni", parameters);

            try
            {
                ViewFactory.Create(parentViewable, spec);
                Assert.IsFalse(true);
            }
            catch (ViewProcessingException ex)
            {
                log.Debug(".testInvalidViewParameters Expected exception caught, msg=" + ex.Message);
            }
        }

        [Test]
        public virtual void testInvalidAttach()
        {
            // View doesn't allow non-numeric fields, the factory should check this
            IList<Object> parameters = new List<Object>();
            parameters.Add(new StringValue("symbol"));
            ViewSpec spec = new ViewSpec("stat", "uni", parameters);

            try
            {
                ViewFactory.Create(parentViewable, spec);
                Assert.IsFalse(true);
            }
            catch (ViewProcessingException ex)
            {
                log.Debug(".testInvalidViewName Expected exception caught, msg=" + ex.Message);
            }
        }

        private static readonly Log log = LogFactory.GetLog(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
    }
}
